igb/repl.go

```diff
@@ -193,31 +224,28 @@ reset:
 				}
 				println(prompt2 + indent(igb.indents) + igb.lines)
 				igb.indents++
-				igb.sm.Event(Waiting)
-				igb.rl.SetPrompt(prompt2 + indent(igb.indents))
 				igb.cmds = append(igb.cmds, igb.lines)
+				igb.rl.SetPrompt(prompt2 + indent(igb.indents))
+				igb.sm.Event(Waiting)
 				igb.caseBlock = true
 				continue
 
 			// To handle such as 'else' or 'elsif'
 			case pErr.IsUnexpectedToken():
-				igb.indents--
-				println(prompt2 + indent(igb.indents) + igb.lines)
-				igb.indents++
-				igb.sm.Event(Waiting)
+				println(prompt2 + indent(igb.indents-1) + igb.lines)
```

Comment:
can you explain the calculation here in comment?

Reply:
I mean the prompt should be `¤` even on the top level indentation when the line is `else` or `elif`

```ruby
» if true
¤   99
¤ else
¤   88
» end
#» 99
```

```ruby
» if true
¤   99
¤ elsif true
¤   88
¤ elsif true
¤   77
¤ else
¤   66
» end
#» 99
```

Reply:
Nice explanation, I hope you can put this in the comment though

Reply:
Commented